packages/ui/src/common/providers/network-endpoints/provider.tsx

```diff
@@ -55,22 +61,18 @@ export const NetworkEndpointsProvider = ({ children }: Props) => {
         setIsLoading(false)
         const errMsg = `Failed to fetch the network configuration from ${configEndpoint}.`
 
-        if (fallbackOnLocalEndpoints) {
-          setEndpoints(localEndpoints)
-          throw new Error(`${errMsg} Falling back on the local endpoints.`)
-        } else {
-          throw new Error(errMsg)
-        }
+        setEndpoints(localEndpoints)
+        throw new Error(`${errMsg} - Falling back on the local endpoints.`)
```

Comment:
This `setEndpoints` was only supposed to happen when `updateNetworkConfig` was called from the `useEffect` at which point the `window.location.reload()` was unnecessary. But when the endpoints are changed after initializing the api and the QN there was sometime some strange and inconsistent behavior between FF and Chrome. So either a reload might be good here or the endpoints could just remain unchanged like in the original `else` branch. WDYT ?

Reply:
Makes sense to just keep the current selection. So I will drop calling setEndpoints(localEndpoints) so avoid need to reload page. I'll test changes overall with Chrome as well since I forgot about this subtle issue.

Reply:
done in [7e017ba](https://github.com/Joystream/pioneer/pull/2532/commits/7e017baf77ef11cac128aa61725bda531b2bfa9b)